activerecord/lib/active_record/persistence.rb

```diff
@@ -67,8 +67,14 @@ def create!(attributes = nil, &block)
       # how this "single-table" inheritance mapping is implemented.
       def instantiate(attributes, column_types = {}, &block)
         klass = discriminate_class_for_record(attributes)
+        instantiate_instance_of(klass, attributes, column_types, &block)
+      end
+
+      # Given a class, an attributes hash, +instantiate+ returns a new instance
```

Comment:
I think it is `+instantiate_instance_of+` instead of `instantiate` here. But I guess we can remove that part of the documentation if we rephrase it.